engine/src/keysinuse_engine.c

```diff
@@ -74,11 +77,67 @@ static int control(ENGINE *e, int cmd, long i, void *p, void (*func)(void))
     return ret;
 }
 
+
+static void  init_once()
+{
+    log_init();
+    // This will read the engine's config explicitly, so that the
+    // engine's config is still used when the engine is loaded
+    // programatically, and not included in the default config
+    // (e.g. on CBL-Mariner). This is also done early, so that
+    // the engine does not bind any functions anything if logging
+    // is disabled.
+    CONF *conf = NCONF_new(NULL);
+    conf = NCONF_new(NULL);
+    if (!NCONF_load(conf, keysinuse_conf_location, NULL))
+    {
+        log_error("Failed to load keysinuse config,OPENSSL_%ld", ERR_get_error());
+        goto err;
+    }
+
+    int i, success;
+    long num_val = 0;
+    char *string_val = NULL;
+    for (i = 0; i < (sizeof(supported_cmds) / sizeof(ENGINE_CMD_DEFN)) && supported_cmds[i].cmd_num != 0; i++)
+    {
+        if (supported_cmds[i].cmd_flags & ENGINE_CMD_FLAG_STRING)
+        {
+            string_val = NCONF_get_string(conf, keysinuse_conf_section, supported_cmds[i].cmd_name);
```

Comment:
>string_val [](http://example.com/codeflow?start=12&length=10)

Can't find documentation for NCONF_get_string - I'm guessing it returns a pointer into a buffer owned by the CONF object, so this string doesn't need to be freed separately?

Reply:
There isn't any docs for 1.1.1 on this API as far as I can tell, but that's correct. The entire config gets freed all at once.